src/shared/util.js

```diff
@@ -92,10 +96,11 @@ export function toString (val: any): string {
 
 /**
  * Convert an input value to a number for persistence.
+ * Check for special case of signed zero.
  * If the conversion fails, return original string.
  */
 export function toNumber (val: string): number | string {
-  const n = parseFloat(val)
+  const n = isNumNegZero(val) ? '-0' : parseFloat(val)
```

Comment:
let's inline the check and remove the function

```suggestion
  const n = 1 / val === -Infinity ? '-0' : parseFloat(val)
```